production/Uploader.py

```diff
@@ -17,26 +17,85 @@ def __init__(self, q=None):
             raise Exception('Uploader.__init__():  missing parameter')
         if not isinstance(q, multiprocessing.queues.Queue):
             raise Exception('Uploader.__init__():  parameter not of type ""')
+        
+    
         self.queue = q
         config = Utility.getProjectConfig()
         # Extract relevant config data
+        inputKey = config.get("dropboxinfo", "key")
+        inputSecret = config.get("dropboxinfo", "secret")
+        inputAccessToken = config.get("dropboxinfo", "accesstoken")
         self.directoryName = config.get('directories', 'imagedirectory')
         os.chdir(self.directoryName)
         # INSERT DROPBOX DATA FROM CONFIG HERE
-    
+        self.myKey = inputKey
+        self.mySecret = inputSecret
+        self.myAccessToken = inputAccessToken
+        self.myClient = None
+        self.clientAccountInfo = None
+        self.setApp(inputKey, inputSecret)
+        self.setAccessToken(inputAccessToken)
+        self.setClient()
+        
     def run(self):
         print "Hi, I'm an Uploader!"
         time.sleep(1)
         while True:
             time.sleep(constants.POLL_TIME)
             print "Uploader, checking in! pid:", os.getpid()
+            if not self.queue.empty():
+                self.uploadBatch()
+    
+    def setApp(self, appKey = None, appSecret = None):
+        self.myKey = appKey
+        self.mySecret = appSecret   
+        return None
```

Comment:
Not a big deal, but if you're writing a method that doesn't return anything, it's not necessary to use 'return None' -- Python handles that.